lib/client/src/client.rs

```diff
@@ -788,6 +898,142 @@ impl MangoClient {
         Ok(ixs)
     }
 
+    async fn deactivate_first_active_unused_serum3_orders(
+        &self,
+        account: &mut MangoAccountValue,
+    ) -> anyhow::Result<PreparedInstructions> {
+        let mut serum3_closable_order_market_index = None;
+
+        for p in account.all_serum3_orders() {
+            let open_orders_acc = self
+                .account_fetcher
+                .fetch_raw_account(&p.open_orders)
+                .await?;
+            let open_orders_bytes = open_orders_acc.data();
+            let open_orders_data: &serum_dex::state::OpenOrders = bytemuck::from_bytes(
+                &open_orders_bytes[5..5 + std::mem::size_of::<serum_dex::state::OpenOrders>()],
+            );
+
+            let is_closable = open_orders_data.free_slot_bits == u128::MAX
+                && open_orders_data.native_coin_total == 0
+                && open_orders_data.native_pc_total == 0;
+
+            if is_closable {
+                serum3_closable_order_market_index = Some(p.market_index);
+                break;
+            }
+        }
+
+        let first_closable_slot =
+            serum3_closable_order_market_index.expect("couldn't find any serum3 slot available");
```

Comment:
`a.expect()` will panic the program if a is None. Prefer to convert this into an error that we can return